src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -322,56 +324,58 @@ public boolean isFullBlock(final IBlockState state)
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
 
-    /**
-     * Convert the BlockState into the correct metadata value.
-     *
-     * @deprecated (Remove this as soon as minecraft offers anything better).
-     */
     @NotNull
     @Override
-    @Deprecated
-    public IBlockState withRotation(@NotNull final IBlockState state, final Rotation rot)
+    protected BlockStateContainer createBlockState()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @Deprecated annotation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1133)